azurerm/internal/services/cosmos/common/indexing_policy.go

```diff
@@ -66,6 +66,29 @@ func expandAzureRmCosmosDBIndexingPolicyCompositeIndexes(input []interface{}) *[
 	return &indexes
 }
 
+func expandAzureRmCosmosDBIndexingPolicySpatialIndexes(input []interface{}) *[]documentdb.SpatialSpec {
+	if len(input) == 0 || input[0] == nil {
+		return nil
+	}
+	indexes := make([]documentdb.SpatialSpec, 0)
+	spatialTypes := []documentdb.SpatialType{
+		documentdb.SpatialTypeLineString,
+		documentdb.SpatialTypeMultiPolygon,
+		documentdb.SpatialTypePoint,
+		documentdb.SpatialTypePolygon,
+	}
+
+	for _, i := range input {
+		indexPair := i.(map[string]interface{})
+		indexes = append(indexes, documentdb.SpatialSpec{
+			Types: &spatialTypes,
+			Path:  utils.String(indexPair["path"].(string)),
+		})
+	}
```

Comment:
should we expose this in schema instead of applying each spatial index to all types?

Reply:
i'm not sure why that doesn't mean we shouldn't expose the four types for people to individually set?

Reply:
sorry but i still don't follow, why are there four "types" to set an index path on if they will all be the same regardless of what we set?

Reply:
I think it's by design by service. Here is the service's reply:
https://gist.github.com/yupwei68/6c3ecffe6aeec33d945210e70bb3513a#gistcomment-3735507

Reply:
my point is would someone ever want to set a different spatial index path for different types? it type1 is path1 and type2 is path2 ect or will there always only ever be one index path for all types?

Reply:
There will always be index paths for all types, according to the service team. The user can't specify the specific types for their paths.

Reply:
ok so to be clear, there is a single path that is shared by all types regardless and its not possible to assign different paths to different types? i assume the portal only has a "path" property then and hides away types from the user? if this is all the case maybe it make sense to document htis in the property so users know the path is applied to all types?

Reply:
Yes. The portal has not supported this yet. I have updated the document and I have marked the types computed.